test/auto-take-profit.ts

```diff
@@ -0,0 +1,564 @@
+                exchangeData.exchangeAddress = ONE_INCH_V4_ROUTER
+                exchangeData._exchangeCalldata = forgeUnoswapCalldata(
+                    hardhatUtils.addresses.WETH,
+                    exchangeData.fromTokenAmount,
+                    exchangeData.minToTokenAmount,
+                )
+            })
+
+            describe('when Trigger is above next col ratio', async () => {
+                let triggerId: number
+                let triggerData: BytesLike
+                let executionData: BytesLike
+                let signer: Signer
+
+                beforeEach(async () => {
+                    // makeSnapshot
+                    snapshotId = await hre.ethers.provider.send('evm_snapshot', [])
+                    signer = await hardhatUtils.impersonate(proxyOwnerAddress)
+
+                    triggerData = encodeTriggerData(
+                        testCdpId,
+                        TriggerType.AUTO_TP_DAI,
+                        nextPrice.add('32665974900000000000'),
+                        1000,
+                    )
+
+                    executionData = generateTpOrSlExecutionData(
+                        MPAInstance,
+                        false, // to collateral
+                        cdpData,
+                        exchangeData,
+                        serviceRegistry,
+                    )
+
+                    // addTrigger
+                    const dataToSupply = AutomationBotInstance.interface.encodeFunctionData('addTrigger', [
+                        testCdpId,
+                        TriggerType.AUTO_TP_DAI,
+                        0,
+                        triggerData,
+                    ])
+                    const tx = await usersProxy.connect(signer).execute(AutomationBotInstance.address, dataToSupply)
+
+                    const txRes = await tx.wait()
+                    const [event] = getEvents(txRes, AutomationBotInstance.interface.getEvent('TriggerAdded'))
+                    triggerId = event.args.triggerId.toNumber()
+                })
+
+                afterEach(async () => {
+                    // revertSnapshot
+                    await hre.ethers.provider.send('evm_revert', [snapshotId])
+                })
+
+                it('should revert trigger execution', async () => {
+                    const tx = AutomationExecutorInstance.execute(
+                        executionData,
+                        testCdpId,
+                        triggerData,
+                        AutoTakeProfitCommandInstance.address,
+                        triggerId,
+                        0,
+                        0,
+                        188000,
+                    )
+                    await expect(tx).to.be.revertedWith('bot/trigger-execution-illegal')
+                })
+            })
+
+            describe('when Trigger is below next col ratio', async () => {
```

Comment:
I smell some copy-pasting from close here, there is no collRatio consideration in Auto take Profit

Reply:
I did the first version using coll ratio, but 100% correct, copy paste from stop loss